src/main/java/io/github/bobocodebreskul/server/WebPathScanner.java

```diff
@@ -0,0 +1,167 @@
+package io.github.bobocodebreskul.server;
+
+import static io.github.bobocodebreskul.context.support.ReflectionUtils.checkIfAnnotationHasAnnotationType;
+import static io.github.bobocodebreskul.context.support.ReflectionUtils.invokeAnnotationMethod;
+import static io.github.bobocodebreskul.server.WebPathValidator.validatePath;
+import static io.github.bobocodebreskul.server.enums.RequestMethod.GET;
+
+import io.github.bobocodebreskul.context.annotations.Delete;
+import io.github.bobocodebreskul.context.annotations.Get;
+import io.github.bobocodebreskul.context.annotations.Head;
+import io.github.bobocodebreskul.context.annotations.Post;
+import io.github.bobocodebreskul.context.annotations.Put;
+import io.github.bobocodebreskul.context.annotations.RequestMapping;
+import io.github.bobocodebreskul.context.annotations.RestController;
+import io.github.bobocodebreskul.context.exception.AmbiguousHttpAnnotationException;
+import io.github.bobocodebreskul.context.exception.DuplicatePathException;
+import io.github.bobocodebreskul.context.registry.BringContainer;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * This class is responsible for scanning and creating web paths. It uses bring container to
+ * retrieve all Beans marked as {@link RestController} then retrieve path and http method from
+ * public methods marked as {@link Post}, {@link Get}, {@link Delete}, {@link Put}, {@link Head},
+ * {@link RequestMapping}
+ */
+@Slf4j
+public class WebPathScanner {
+
+  private final BringContainer container;
+
+  /**
+   * Constructs a new instance of {@code WebPathScanner} with the specified bring container.
+   *
+   * @param container The container providing information about controllers.
+   */
+  public WebPathScanner(BringContainer container) {
+    this.container = container;
+  }
+
+  private static String getPrefixPath(Class<?> controllerClass) {
+    return controllerClass.isAnnotationPresent(RequestMapping.class) ?
+        controllerClass.getAnnotation(RequestMapping.class).value() : "";
+  }
+
+  private static Annotation getHttpAnnotation(Method method) {
+    var annotationList = Arrays.stream(method.getAnnotations())
+        .filter(
+            annotation -> checkIfAnnotationHasAnnotationType(annotation, RequestMapping.class))
+        .toList();
+
+    if (annotationList.isEmpty()) {
+      return null;
```

Comment:
There is `Optional` class for cases when `null` could be return value, it's much more better than possible NPE.

Reply:
@hladchenko @JoshukLite in java 17 were introduced useful nullpoiner exception https://openjdk.org/jeps/358 so you will easily find this null pointer in your code. This method will not be used outside of this class so it is the only place where it will be used, besides that you should remember that Optional is wrapper so it will take additional resources and time create the wrapper.